### PR TITLE
ENG-10012: Remove rewrite_on_violation, add enable_dataset_rewrites

### PR DESCRIPTION
**Pull request by Yowgf**
## Description of the change

This PR renames parameter `rewrite_on_violation` in resource `cyral_repository_conf_analysis` to `enable_dataset_rewrites`.

This PR also corrects some v3 documentation.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] Jira issue referenced in commit message and/or PR title

### Testing

All Acceptance tests are passing.

### CODE REVIEW

File: cyral/resource_cyral_repository_conf_analysis.go [hunk 1]
```diff
@@ -25,10 +25,20 @@ type UserFacingConfig struct {
...
 	LogGroups                  []string `json:"logGroups,omitempty"`
 	Redact                     string   `json:"redact"`
-	RewriteOnViolation         bool     `json:"rewriteOnViolation"`
+
+	// TODO: RewriteOnViolation is deprecated and only works for CP versions
+	// <= 3.0.1. In 3.0.2, it was substituted by
+	// `enableDatasetRewrites`. Remove this in the next MAJOR release.
+	RewriteOnViolation bool `json:"rewriteOnViolation"`
+
+	EnableDatasetRewrites bool `json:"enableDatasetRewrites"`
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_conf_analysis.go:34]:
```suggestion
	EnableDatasetRewrites bool `json:"enableDatasetRewrites"`
```

File: cyral/resource_cyral_repository_conf_analysis.go [hunk 2]
```diff
@@ -25,10 +25,20 @@ type UserFacingConfig struct {
...
 }
 
 func repositoryConfAnalysisResourceSchemaV0() *schema.Resource {
+	rewriteOnViolationDeprecationMessage := "This argument only works with " +
+		"control plane versions up to 3.0.1. It will be removed in a " +
+		"future release. Use `enable_dataset_rewrites` instead."
+
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_conf_analysis.go:41]:
```suggestion
```

File: cyral/resource_cyral_repository_conf_analysis.go [hunk 3]
```diff
@@ -77,10 +87,18 @@ func repositoryConfAnalysisResourceSchemaV0() *schema.Resource {
 				Type:        schema.TypeBool,
 				Optional:    true,
 			},
+			// TODO: Remove `rewrite_on_violation` next MAJOR
+			// release.
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_conf_analysis.go:91]:
```suggestion
```

File: cyral/resource_cyral_repository_conf_analysis.go [hunk 4]
```diff
@@ -77,10 +87,18 @@ func repositoryConfAnalysisResourceSchemaV0() *schema.Resource {
...
 			"rewrite_on_violation": {
 				Description: "If set to `true` it will enable rewriting queries on violations.",
 				Type:        schema.TypeBool,
 				Optional:    true,
+				Deprecated:  rewriteOnViolationDeprecationMessage,
+			},
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_conf_analysis.go:97]:
```suggestion
```

File: cyral/resource_cyral_repository_conf_analysis.go [hunk 5]
```diff
@@ -281,6 +299,12 @@ func getConfAnalysisDataFromResource(d *schema.ResourceData) (RepositoryConfAnal
 		}
 	}
 
+	// To avoid breaking backwards compatibility with previous control plane
+	// versions, we provide both values to the API.
+	deprecatedRewriteOnViolationArgVal := d.Get("rewrite_on_violation").(bool)
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_conf_analysis.go:304]:
```suggestion
```

File: cyral/resource_cyral_repository_conf_analysis.go [hunk 6]
```diff
@@ -291,7 +315,11 @@ func getConfAnalysisDataFromResource(d *schema.ResourceData) (RepositoryConfAnal
 			CommentAnnotationGroups:    annotationGroups,
 			LogGroups:                  logGroups,
 			Redact:                     d.Get("redact").(string),
-			RewriteOnViolation:         d.Get("rewrite_on_violation").(bool),
+
+			// Remove RewriteOnViolation in the next MAJOR release.
+			RewriteOnViolation: enableDatasetRewrites,
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_conf_analysis.go:320]:
```suggestion
```

File: cyral/resource_cyral_repository_conf_analysis.go [hunk 7]
```diff
@@ -317,5 +345,9 @@ func setConfAnalysisDataToResource(d *schema.ResourceData, resourceData Reposito
 	d.Set("enable_data_masking", resourceData.Config.EnableDataMasking)
 	d.Set("log_groups", logGroupsSet)
 	d.Set("redact", resourceData.Config.Redact)
+
+	// TODO: Remove RewriteOnViolation on next MAJOR release.
 	d.Set("rewrite_on_violation", resourceData.Config.RewriteOnViolation)
+
```
Comment (wcmjunior) [inline cyral/resource_cyral_repository_conf_analysis.go:351]:
```suggestion
```

File: docs/resources/repository_conf_analysis.md [hunk 8]
```diff
@@ -79,7 +80,7 @@ resource "cyral_repository_conf_analysis" "all_conf_analysis_disabled" {
   - `new-connections` - Log new connections.
   - `closed-connections` - Log closed connections.
 - `redact` (String) Valid values are: `all`, `none` and `watched`. If set to `all` it will enable the redact of all literal values, `none` will disable it, and `watched` will only redact values from tracked fields set in the Datamap.
-- `rewrite_on_violation` (Boolean) If set to `true` it will enable rewriting queries on violations.
+- `rewrite_on_violation` (Boolean, Deprecated) If set to `true` it will enable rewriting queries on violations.
```
Comment (wcmjunior) [inline docs/resources/repository_conf_analysis.md:83]:
```suggestion
```

**Review by wcmjunior — COMMENTED** [review]
Please remove the old field entirely. This is a bug as this is no longer supported since CP `v3.x`.

**Comment by Yowgf** [comment]
@wcmjunior could you please take another look? I have added some changes in the migration guide and in the CHANGELOG file, because I think we were not being complete in terms of notifying breaking changes. Please LMK your thoughts.

File: CHANGELOG.md [hunk 9]
```diff
@@ -10,6 +10,24 @@ Resources incompatible with Control Planes `v2.x`:
...
+- `auth0_domain`
+- `auth0_client_id`
+- `auth0_client_secret`
+
+Renamed resource arguments:
+
+- `cyral_policy_rule.id` -- now
+  `cyral_policy_rule.policy_rule_id`. `cyral_policy_rule.id` now contains a
+  composed ID in the format `{policy_id}/{policy_rule_id}`.
+
```
Comment (wcmjunior) [inline CHANGELOG.md:30]:
```suggestion
```

**Review by wcmjunior — APPROVED** [review]

**Pull request merged**